src/Spec.Reporter.savi

```diff
@@ -43,14 +48,35 @@
   :fun ref spec_ended(spec String)
     @env.print_err("") // just a newline
 
+  :fun ref overall_began
+    @_overall_began = Time.Measure.current_milliseconds
+    None
+
+  :fun ref overall_ended
+    elapsed_s = (Time.Measure.current_milliseconds - @_overall_began) / 1000
+
+    msg = case (
+    | @_num_failed_assertions > 0 | @_red("FAIL")
+    |                               @_green("OK")
+    )
+    num_passed = @_num_total_assertions - @_num_failed_assertions
+
+    @env.print_err("test result: \(--msg). \(num_passed) passed; \(@_num_failed_assertions) failed; finished in \(elapsed_s.format.decimal)s")
+    None
+
   :fun ref example_began(spec String, example String)
+    @failed_assertions.clear
     @env.write_err("  \(example) ")
 
   :fun ref example_ended(spec String, example String, stats Spec.Statistics.ForExample)
+    @_num_total_examples = @_num_total_examples + 1
```

Comment:
This variable is not being used anywhere, right?

It should probably be removed unless you want to print it.

Reply:
maybe we want to print the number of examples which corresponds to the number of test cases. not just the number of assertions.

Reply:
I think printing the number of examples would be fine. I'm just thinking there's no reason to gather that info unless we're printing it.

Or were you thinking that users would override the printing behavior themselves and make use of this statistic?

Reply:
good point. I am removing it for now and merge. once we want to print it out, we can easily add it again.

No, I don't want to make it too much configurable. If the user needs statistics, we would better write a JSON Reporter.